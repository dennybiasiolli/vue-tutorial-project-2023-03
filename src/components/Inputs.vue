<script setup>
import { ref } from 'vue'

const message = ref('')
const checked = ref(false)

const people = [
  { id: 1, name: 'Jack' },
  { id: 2, name: 'John' },
  { id: 3, name: 'Mike' },
]
const checkedNames = ref([])

const picked = ref('')
const selected = ref(null)
const muliSelected = ref([])

const num = ref(0)
const trimmedText = ref('')
</script>

<template>
  <h2>Inputs</h2>

  <h4>Text</h4>
  <p>Message is: {{ message }}</p>
  <input v-model="message" placeholder="edit me" />
  <p>Message (lazy) is: {{ message }}</p>
  <input v-model.lazy="message" placeholder="edit me" />
  <br>
  <span>Multiline message is:</span>
  <p style="white-space: pre-line;">{{ message }}</p>
  <textarea v-model="message" placeholder="add multiple lines"></textarea>

  <h4>Checkbox</h4>
  <input type="checkbox" id="checkbox" v-model="checked" />
  <label for="checkbox">{{ checked }}</label>

  <div>Checked names: {{ checkedNames }}</div>
  <template v-for="person in people" :key="person.id">
    <input type="checkbox" :id="person.id" :value="person.id" v-model="checkedNames">
    <label :for="person.id">{{ person.name }}</label>
  </template>

  <h4>Radio</h4>
  <div>Picked: {{ picked }}</div>
  <template v-for="person in people" :key="person.id">
    <input type="radio" :id="person.id" :value="person.id" v-model="picked" />
    <label :for="person.id">{{ person.name }}</label>
  </template>

  <h4>Select</h4>
  <div>Selected: {{ selected }}</div>
  <select v-model="selected">
    <option disabled :value="null">Please select one</option>
    <option
      v-for="person in people"
      :key="person.id"
      :value="person"
    >{{ person.name }}</option>
  </select>

  <div>Selected: {{ muliSelected }}</div>
  <select v-model="muliSelected" multiple>
    <option
      v-for="person in people"
      :key="person.id"
      :value="person"
    >{{ person.name }}</option>
  </select>

  <div>
    <p>Number is: {{ num }} --&gt; {{ num === 1 }}</p>
    <input type="number" v-model="num" placeholder="edit me" />
    <input type="text" v-model.number="num" placeholder="edit me" />
  </div>

  <div>Trimmed text is &quot;{{ trimmedText }}&quot;</div>
  <input v-model.trim="trimmedText" />
</template>
