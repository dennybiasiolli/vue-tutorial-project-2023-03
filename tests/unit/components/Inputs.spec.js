import { shallowMount } from '@vue/test-utils'
import { expect, it } from 'vitest'
import Inputs from '@/components/Inputs.vue'

it('renders as expected', () => {
  const wrapper = shallowMount(Inputs)
  expect(wrapper.html()).toMatchSnapshot()
})
