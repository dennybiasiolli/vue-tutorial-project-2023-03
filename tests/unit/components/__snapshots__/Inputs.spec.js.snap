// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders as expected 1`] = `
"<h2>Inputs</h2>
<h4>Text</h4>
<p>Message is: </p>
<input placeholder=\\"edit me\\">
<p>Message (lazy) is: </p>
<input placeholder=\\"edit me\\">
<br>
<span>Multiline message is:</span>
<p style=\\"white-space: pre-line;\\"></p>
<textarea placeholder=\\"add multiple lines\\"></textarea>
<h4>Checkbox</h4>
<input type=\\"checkbox\\" id=\\"checkbox\\">
<label for=\\"checkbox\\">false</label>
<div>Checked names: []</div>
<input type=\\"checkbox\\" id=\\"1\\" value=\\"1\\">
<label for=\\"1\\">Jack</label>
<input type=\\"checkbox\\" id=\\"2\\" value=\\"2\\">
<label for=\\"2\\">John</label>
<input type=\\"checkbox\\" id=\\"3\\" value=\\"3\\">
<label for=\\"3\\">Mike</label>
<h4>Radio</h4>
<div>Picked: </div>
<input type=\\"radio\\" id=\\"1\\" value=\\"1\\">
<label for=\\"1\\">Jack</label>
<input type=\\"radio\\" id=\\"2\\" value=\\"2\\">
<label for=\\"2\\">John</label>
<input type=\\"radio\\" id=\\"3\\" value=\\"3\\">
<label for=\\"3\\">Mike</label>
<h4>Select</h4>
<div>Selected: </div>
<select>
  <option disabled=\\"\\">Please select one</option>
  <option value=\\"[object Object]\\">Jack</option>
  <option value=\\"[object Object]\\">John</option>
  <option value=\\"[object Object]\\">Mike</option>
</select>
<div>Selected: []</div>
<select multiple=\\"\\">
  <option value=\\"[object Object]\\">Jack</option>
  <option value=\\"[object Object]\\">John</option>
  <option value=\\"[object Object]\\">Mike</option>
</select>
<div>
  <p>Number is: 0 --> false</p><input type=\\"number\\" placeholder=\\"edit me\\"><input type=\\"text\\" placeholder=\\"edit me\\">
</div>
<div>Trimmed text is \\"\\"</div>
<input>"
`;
